### ISSUE TITLE
Add unit test that verifies persistent storage has been created

### ISSUE THREAD

**Issue by Therzok** [mono/monodevelop#4185]
> VS bug [#583994](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/583994)

**Comment by Therzok** [comment]
Waiting for test-split to avoid conflicts. Sample test code:
```
[Test]
		public async Task TestWorkspacePersistentStorage ()
		{
			//Initialize IdeApp so IdeApp.Workspace is not null
			if (!IdeApp.IsInitialized)
				IdeApp.Initialize (new ProgressMonitor ());
			string solFile = Util.GetSampleProject ("console-project", "ConsoleProject.sln");
			var tcs = new TaskCompletionSource<Solution> ();
			IdeApp.Workspace.SolutionLoaded += (s, e) => {
				tcs.SetResult (e.Solution);
			};

			await IdeApp.Workspace.OpenWorkspaceItem (solFile);
			var sol = await tcs.Task;
			var ws = await TypeSystemService.GetWorkspaceAsync (sol);

			foreach (var projectFile in sol.GetAllProjects ().SelectMany (x => x.Files.Where (file => file.BuildAction == BuildAction.Compile))) {
				var projectId = ws.GetProjectId (projectFile.Project);
				var docId = ws.GetDocumentId (projectId, projectFile.Name);
				var doc = ws.GetDocument (docId);
				if (!doc.SupportsSyntaxTree)
					continue;

				await Microsoft.CodeAnalysis.FindSymbols.SyntaxTreeIndex.PrecalculateAsync (doc, CancellationToken.None);
			}

			var storageLocationService = ws.Services.GetService<IPersistentStorageLocationService> ();
			var storageLocation = System.IO.Path.Combine (
				storageLocationService.TryGetStorageLocation (ws.CurrentSolution.Id),
				"sqlite3",
				"storage.ide");

			var fi = new System.IO.FileInfo (storageLocation);
			Assert.That (fi.Length, Is.GreaterThan (0));
		}
```

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-12] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Author: Marius Ungureanu  
 Commit: a26584b47ef1eacd5db9f824d3cbc433d2fb855d (xamarin/monodevelop)
